### PR TITLE
T0yv0/add labels test matrix

### PR DESCRIPTION
**Pull request by iwahbe**
This is a reviewable copy of https://github.com/pulumi/pulumi-gcp/pull/1309.

### CODE REVIEW

File: examples/examples_go_test.go [hunk 1]
```diff
@@ -19,6 +18,8 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
```
Comment (iwahbe) [inline examples/examples_go_test.go:22]: `"os"` should be in the top block of imports.

**Comment by t0yv0** [comment]
Feel free to edit my PR as in change base etc.

File: examples/examples_go_test.go [hunk 2]
```diff
@@ -179,3 +180,178 @@ func GenerateRandomString(n int) (string, error) {
...
+				Labels:        label2,
+			})
+		}
+	}
+
+	t.Logf("total state space: %v states", len(states))
+	t.Logf("random-sampling 100 state transitions")
+
+	for i := 0; i < 100; i++ {
+		t.Run(fmt.Sprintf("test-%d", i), func(t *testing.T) {
```
Comment (iwahbe) [inline examples/examples_go_test.go:263]: `""` is interpreted as `fmt.Sprintf("#%d", i)`
```suggestion
		t.Run("", func(t *testing.T) {
```

**Review by iwahbe — COMMENTED** [review]
2 style nits, but otherwise this looks great.

File: examples/examples_go_test.go (hunk 1 above)
Comment (t0yv0) [reply examples/examples_go_test.go:22]: Have you automated ordering imports yet? I use C-c C-a for go-import-add but it does not do the right thing. On save I use gofmt. Perhaps there's something stronger than gofmt that could reorder them the way we need to on save?

Comment (iwahbe) [reply examples/examples_go_test.go]: I have automated sorting them within blocks, but I have not automated sorting between blocks. I generally add imports manually, sometimes `eglot` does automatic insertion. I run `gofmt` on save, but that only alphabetizes, not sorting into blocks.

Comment (iwahbe) [reply examples/examples_go_test.go]: https://github.com/mvdan/gofumpt is the stronger version. You could try `(setq gofmt-command "gofumpt")`.

**Pull request closed without merging**